cowamm/kpi/competitors/curve/curve_kpis_4232873.sql

```diff
@@ -0,0 +1,27 @@
+-- Computes volume, tvl and APR for Curve pools
+-- APR is measured as the fees earned per $ invested, over the last 24 hours, projected over 1 year
+-- Parameters:
+-- {{blockchain}}: The blockchain to query
+-- {{competitor_end_time}}: The end time of the time window (end_time - 1 day; end_time), defaults to now()
+select
+    r1.contract_address,
+    r1.fee,
+    r1.tvl,
+    sum(amount_usd) as volume,
+    365 * sum(amount_usd * r.fee / r.tvl) as apr
```

Comment:
Thanks for the explanation, I get it now. Just as a comment (not actionable assuming performance of the join isn't an issue), but this _could_ also be computed using a "rolling sum" which would avoid the self join.

Something like

```
select
    contract_address,
    fee,
    tvl,
    sum(amount_usd) over (partition by contract_address order by latest_per_pool) as volume,
    365 * sum(amount_usd * fee / tvl) over (partition by contract_address order by latest_per_pool) as apr
```

Reply:
There is indeed a performance issue overall for all queries. So I implemented it. Result: it divides by 2 the execution time